src/stacks-hierarchy/Root_PurchasePaymentScreen/Root_PurchasePaymentScreen.tsx

```diff
@@ -51,6 +52,17 @@ export const Root_PurchasePaymentScreen = ({route, navigation}: Props) => {
     reserveMutation.data?.recurring_payment_id,
   );
 
+  useEffect(() => {
+    const orderId = reserveMutation.data?.order_id;
+    if (orderId) {
+      const allPossibleFareContracts = [...(fareContracts ?? []), ...(sentFareContracts ?? [])];
+      const fareContractReceived = allPossibleFareContracts.some((fc) => fc.orderId === orderId);
+      if (fareContractReceived) {
+        navigateToActiveTicketsScreen();
+      }
+    }
+  }, [fareContracts, sentFareContracts, reserveMutation.data?.order_id, navigateToActiveTicketsScreen]);
```

Comment:
One general tip about `useEffect` is that they are often a little simpler to read if transformation/calculation is done outside of it. This also gives fewer dependencies, which decreases the chance of bugs.

Here it would be something like:
```tsx
const allFareContracts = [...(fareContracts ?? []), ...(sentFareContracts ?? [])];
const fareContractReceived = allFareContracts.some((fc) => 
    fc.orderId === reserveMutation.data?.order_id);

useEffect(() => {
  if (fareContractReceived) {
    navigateToActiveTicketsScreen();
  }
}, [fareContractReceived, navigateToActiveTicketsScreen])
```

Regarding doing the `.some` on every render, one can consider to use `useMemo`. But in most cases, a [simple iteration is not expensive](https://react.dev/learn/you-might-not-need-an-effect#how-to-tell-if-a-calculation-is-expensive).

Reply:
Good suggestion and made fixes based on those.